### PR TITLE
Vulkan video

### PR DESCRIPTION
**Pull request by AWoloszyn**

### CODE REVIEW

File: gapii/cc/spy.cpp [hunk 1]
```diff
@@ -108,6 +108,9 @@ const uint32_t kStartMidExecutionCapture =  0xdeadbeef;
 const int32_t  kSuspendIndefinitely = -1;
 
 const uint8_t kCoreAPI = 0;
```
Comment (ben-clayton) [inline gapii/cc/spy.cpp]: Not too happy with these constants needing to match those in the API file.
At the very least please can we have a comment?

Comment (AWoloszyn) [inline gapii/cc/spy.cpp]: Agreed. I will add a comment right now.
I haven't figure out a good way to handle this. We could maybe generate some relatively empty header files that we can include without having to deal with weird include cycles.

Comment (AWoloszyn) [inline gapii/cc/spy.cpp]: Actually come to think of it, I might be able to emit something into vulkan_spy.h / gles_spy.h and handle it that way. I will give it a shot.

File: gapis/gfxapi/templates/api_spy.h.tmpl [hunk 2]
```diff
@@ -62,6 +62,8 @@
 ¶
     {{ForEach (AllCommands $) "DeclareCommand" | JoinWith "\n"}}
 ¶
+    const static uint32_t kApiIndex = {{$.Index}};
```
Comment (ben-clayton) [inline gapis/gfxapi/templates/api_spy.h.tmpl]: Awesome, thank you!

**Pull request merged**